contracts/TelfordL1Relayer.sol

```diff
@@ -0,0 +1,135 @@
+import "arb-bridge-eth/contracts/bridge/Inbox.sol";
+import "./TelfordSource.sol";
+
+/**
+ * @dev L1Relayer is responsible for accepting the message from the L2BridgeDest contract on Optimism
+ * and sending (relaying) a message to the TelfordSource contract on Arbitrum
+ */
+
+contract L1Relayer {
+    /* ========== State ========== */
+
+    address public bonderAddress;
+    address public userAddress;
+    uint256 public amount;
+    uint256 public transferId;
+    address private owner;
+    IInbox public inboxArbitrum =
+        IInbox(0x578BAde599406A8fE3d24Fd7f7211c0911F5B29e); // Address of the Arbitrum Inbox on the L1 chain
+    address public crossDomainMessengerOptimism =
+        0x4361d0F75A0186C05f971c566dC6bEa5957483fD;
+    ICrossDomainMessenger public crossDomainMessenger =
+        ICrossDomainMessenger(crossDomainMessenger);
+
+    /* ========== Constants ========== */
+
+    address public constant TELFORD_SOURCE =
+        0xD2dB8075693aA6A0D5C026cdf319D517516c1D40; // Address of Telford source on the testnet
+    address public TELFORD_DESTINATION =
+        crossDomainMessenger.xDomainMessageSender(); // Will update it to hardcode
+
+    /* ========== Events ========== */
+
+    event updatedInfo(
+        address bonder,
+        address user,
+        uint256 amount,
+        uint256 transferId
+    );
+    event RetryableTicketCreated(uint256 indexed ticketId);
+
+    /* ========== Modifier / Constructor ========== */
+
+    constructor() public {
+        owner = msg.sender;
+    }
+
+    modifier onlyOwner() {
+        require(msg.sender == owner, "Only owners can perform this operation!");
+        _;
+    }
+
+    modifier onlyTelfordDestination() {
+        require(
+            msg.sender == crossDomainMessengerOptimism &&
+                crossDomainMessenger.xDomainMessageSender() ==
+                TELFORD_DESTINATION,
+            "Only the Telford Destination Contract can perform this operation!"
+        );
+        _;
+    }
+
+    /* ========== Receiving Function ========== */
+
+    /**
+     * @dev Function that is called on by Optimism's messagers contract via Telford's destination contract
+     * @param _bonder The address of the bonder who provided liquidity for the transaction
+     * @param _user The address of the user who is bridging
+     * @param _amount The amount the user is bridging (sub fees for the bonder)
+     * @param _transferId Number to keep track of the transfer in progress
+     */
+
+    function receiveDestinationTransferConfirmation(
+        address _bonder,
+        address _user,
+        uint256 _amount,
+        uint256 _transferId
+    ) external onlyTelfordDestination {
+        bonderAddress = _bonder;
+        userAddress = _user;
+        amount = _amount;
+        transferId = _transferId;
+        emit updatedInfo(_bonder, _user, _amount, _transferId);
+        relayToArbitrum(10000000000000000, 0, 0);
```

Comment:
I'm going to have to take you up on that because I'm not sure what you're talking about haha. I just sent those numbers to the numbers you said in the chat one time.